### ISSUE TITLE
[Data Issue]: Difference between production and consumption implies unexpectedly high electricity imports for Germany

### ISSUE THREAD

**Issue by w-flo** [electricitymaps/electricitymaps-contrib#6323]
### When did this happen?

Year 2023

### What zones are affected?

DE-LU

### What is the problem?

Electricitymaps displays German electricity production in 2023 as 448 TWh, and consumption as 500 TWh – implying German net imports to be 52 TWh.

[Energy-Charts](https://www.energy-charts.info/charts/energy/chart.htm?l=en&c=DE&interval=year&year=-1&source=entsoe&legendItems=011111111111011111111) displays production according to ENTSO-E of 437,3 TWh. Import balance is given as only 8,6 TWh.

I think that discrepancy is a bit odd and should be investigated.

**Comment by mirkoschaefer** [comment]
Probably two reasons:
- Energy-Charts runs different corrections over their data to improve the data published by ENTSO-E. The corrections are mostly based on reported statistical data (monthly or yearly), but currently unfortunately not transparent.
- Electricitymaps does not include such corrections, since these are country-specific corrections, which would need to be done for each country separately. But to apply the flow tracing algorithm, crossborder physical flows, generation and load need to be balanced (basically, physical inflow plus generation must be equal to physical outflow minus load). I don't know how this is done exactly currently, but the hourly imbalances, which need to be corrected, can be quite large. For Germany, in 2022, the difference between (physical inflows minus physical outflows) and (generation minus load) on a hourly basis in some hours were up to 10 GW.

**Comment by w-flo** [comment]
Energy-Charts now has a new "ENTSO-E" mode in addition to the old "Public" and "Total" modes. After selecting it, it says "Original data ENTSO-E". I guess they don't use these statistical corrections when selecting that mode.

However, the "import balance" values seem unaffected by that option, so maybe these values are not actually original ENTSO-E data even when selecting that mode.

Still, I wonder where electricitymaps gets that "500 TWh of consumption" data point from. Would summing up cross border physical flows into and out of Germany really result in a net inflow of 52 TWh for 2023? That's 6x the energy-charts number of 8.6 TWh. I've compared a few countries and hours on 2023-11-01 and the number from ENTSO-E transparency platform seems to match the energy-charts number perfectly in all these cases.

**Comment by VIKTORVAV99** [comment]
Not sure how important this is to the issue but we don't use the DE-LU bidding zone for the data and just the DE country zone. Could potentially explain some of the difference in it since LU seems to always be importing from DE.

However I have not had time to calculate the numbers.

**Comment by mirkoschaefer** [comment]
> Energy-Charts now has a new "ENTSO-E" mode in addition to the old "Public" and "Total" modes. After selecting it, it says "Original data ENTSO-E". I guess they don't use these statistical corrections when selecting that mode.
> 
> However, the "import balance" values seem unaffected by that option, so maybe these values are not actually original ENTSO-E data even when selecting that mode.
> 
> Still, I wonder where electricitymaps gets that "500 TWh of consumption" data point from. Would summing up cross border physical flows into and out of Germany really result in a net inflow of 52 TWh for 2023? That's 6x the energy-charts number of 8.6 TWh. I've compared a few countries and hours on 2023-11-01 and the number from ENTSO-E transparency platform seems to match the energy-charts number perfectly in all these cases.

Yes, you are right, the 8.6 TWh actually refer to the annual net import based on cross-border physical flows for Germany, just checked using the SMARD data (based on scheduled commercial flows it is 11.7 TWh, but these values have a different origin). So the difference indeed seems quite high, even if you would assume some corrections to align with physical flow balances.

**Comment by w-flo** [comment]
> Not sure how important this is to the issue but we don't use the DE-LU bidding zone for the data and just the DE country zone. Could potentially explain some of the difference in it since LU seems to always be importing from DE.

Oh, sorry for the confusion! I didn't realize you refer to "DE" as a "country zone". This issue is purely about the country Germany, without taking Luxembourg into account.

The issue template asked me for a zone, and as far as I was aware, there is only a "DE-LU" zone and not a "DE" zone, so I entered "DE-LU". However, I was only referring to German electricity imports/exports and consumption/production. I will change it to "DE" since that is apparently a valid zone in this context.
